### PR TITLE
add msal extensions code to main repo

### PR DESCRIPTION
**Pull request by MarkZuber**
Adding in microsoft.identity.client.extensions.msal code from the extensions repo directly into the msal project.  brought over unit tests as well.  will need to evaluate bringing over a few console/manual test apps from there which will be in a separate PR.  

the build infra to create/sign/publish nuget packages is not wired up yet, that will come with the YAML build updates in a separate PR.  this is just to get the code in place in the solution files.

### CODE REVIEW

**Comment by bgavrilMS** [comment]
Assuming the files are all copied, there is nothing to review really :)

File: tests/Microsoft.Identity.Client.Extensions.Msal.UnitTests/MacKeyChainTests.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Identity.Client.Extensions.Msal.UnitTests;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Xunit.Sdk;
+
+namespace Microsoft.Identity.Client.Extensions.Msal
+{
+    [TestClass]
+    public class MacKeyChainTests
```
Comment (henrik-me) [inline tests/Microsoft.Identity.Client.Extensions.Msal.UnitTests/MacKeyChainTests.cs]: assuming these tests doesn't really run today as we don't run the utests on mac. Also assuming that Mac stuff is not "released" - perhaps at least a comment should be added about the mac state?

**Comment by henrik-me** [comment]
Generally agree that we assume the code is great as it has already been reviewed, however all the usual standards should now be enforced at least when making changes to these files. Analyzers, and test coverage. That is our test pass should include manual testing. For now assuming that is happening as the manual tests are also moved?

---
In reply to: [527900463](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1369#issuecomment-527900463) [](ancestors = 527900463)

File: tests/Microsoft.Identity.Client.Extensions.Msal.UnitTests/MacKeyChainTests.cs (hunk 1 above)
Comment (MarkZuber) [reply tests/Microsoft.Identity.Client.Extensions.Msal.UnitTests/MacKeyChainTests.cs]: Sure.  Yes, the mac tests don't run on our environment, but they can be run if you have a mac.  The other tests pass.  I'll add a comment about the status of the mac (and linux) code from a testing and functionality standpoint.

**Comment by MarkZuber** [comment]
> Generally agree that we assume the code is great as it has already been reviewed, however all the usual standards should now be enforced at least when making changes to these files. Analyzers, and test coverage. That is our test pass should include manual testing. For now assuming that is happening as the manual tests are also moved?
> 
> In reply to: [527900463](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1369#issuecomment-527900463) [](ancestors = 527900463)

I'm going to get this in to start working on the build.  In parallel, I'll be evaluating and bringing over the manual tests apps so that when we're ready to actually release these bits out of this repo everything is in place.  I'm just doing it step-wise.

**Comment by jennyf19** [comment]
Does this have documentation?

**Comment by MarkZuber** [comment]
> Does this have documentation?

There is xml documentation on most of this code, but there's nothing in docs.ms at this time.  This library will be released in -preview mode (as it is today in the other repo) so we will want to build up documentation as we finalize our release and support plans for this code.  This code should allow us a centralized place to do further work and finalization on the desktop persistence story.

**Pull request merged**